app/controllers/comments_controller.rb

```diff
@@ -1,5 +1,6 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:create]
 
```

Comment:
If you add this to the top of your file (e.g. line 4):
```ruby
  before_action { authorize(@cohort || Cohort) }
```

You can remove all other instances of `authorize(@comment`. This works for other controllers, too.